src/HDF5Msg.chpl

```diff
@@ -24,1759 +26,1621 @@ module HDF5Msg {
-                }
-                
-                if !endsWithCompleteString[idx] {
-                    shuffleRightIndices[idx] = adjLastSeg;
-                } else {
-                    shuffleRightIndices[idx] = -1;
+                var dims: [0..#1] C_HDF5.hsize_t; // Only rank 1 for now
+
+                // Read array length into dims[0]
+                C_HDF5.HDF5_WAR.H5LTget_dataset_info_WAR(file_id, dsetName.c_str(), 
+                                           c_ptrTo(dims), nil, nil);
+                lengths[i] = dims[0]: int;
+                if lengths[i] == 0 {
+                    skips.add(filename);
+                    h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                        "Adding filename:%s to skips, dsetName:%s, dims[0]:%t".format(filename, dsetName, dims[0]));
                 }
-            
-                if shuffleLeftIndices[idx] > -1 || shuffleRightIndices[idx] > -1 {
-                    /*
-                    * If either of the indices are > -1, this means there's 2..n null characters
-                    * in the char array, which means the char array contains 2..n strings and/or
-                    * string portions.
-                    */   
-                    isSingleString[idx] = false;
+
+            } catch e: Error {
+                throw getErrorWithContext(
+                             msg="in getting dataset info %s".format(e.message()),
+                             lineNumber=getLineNumber(), 
+                             routineName=getRoutineName(), 
+                             moduleName=getModuleName(), 
+                             errorClass='WriteModeError'
+                );
+            }
+        }
+        // Compute subdomain of master array contained in each file
+        var subdoms: [FD] domain(1);
+        var offset = 0;
+        for i in FD {
+            subdoms[i] = {offset..#lengths[i]};
+            offset += lengths[i];
+        }
+        return (subdoms, (+ reduce lengths), skips);
+    }
+
+    /* 
+        Write data from HDF5 dataset into a distributed array.
+        This function gets called when A is a BlockDist or DefaultRectangular array. 
+    */
+    proc read_files_into_distributed_array(A, filedomains: [?FD] domain(1), 
+                                                 filenames: [FD] string, dsetName: string, skips: set(string)) throws 
+        where (MyDmap == Dmap.blockDist || MyDmap == Dmap.defaultRectangular)
+    {
+        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                "entry.a.targetLocales() = %t".format(A.targetLocales()));
+        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                "Filedomains: %t".format(filedomains));
+        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                "skips: %t".format(skips));
+
+        coforall loc in A.targetLocales() do on loc {
```

Comment:
This is where one of the missing localize calls was lost. The `coforall` over `filenames.domain` and then the subsequent indexing into filenames was preserving the locality. This is definitely the preferred method over what I've opened in https://github.com/Bears-R-Us/arkouda/pull/1879, but I won't have a chance to do a more thorough look through this to see if there are other similar things that need to be updated.